scan-command/src/test/java/io/ballerina/scan/internal/ScanCmdTest.java

```diff
@@ -250,4 +248,94 @@ void testScanCommandGenerateScanReportMethodWhenIssuePresent() throws IOExceptio
+                .replace(WINDOWS_LINE_SEPARATOR, LINUX_LINE_SEPARATOR);
+        Assert.assertEquals(result, expected);
+    }
+
+    @Test(description = "test method for sorting static code analysis rules in specified order")
+    void testSorRules() {
+        List<Rule> rules = new ArrayList<>(List.of(
+                RuleFactory.createRule(1, "rule 1", RuleKind.CODE_SMELL, BALLERINA_ORG, "exampleModule"),
+                RuleFactory.createRule(3, "rule 3", RuleKind.BUG, BALLERINAX_ORG, "exampleModule"),
+                RuleFactory.createRule(2, "rule 2", RuleKind.VULNERABILITY, "wso2", "exampleModule"),
+                RuleFactory.createRule(3, "rule 3", RuleKind.BUG),
+                RuleFactory.createRule(1, "rule 1", RuleKind.CODE_SMELL, "exampleOrg", "exampleModule"),
+                RuleFactory.createRule(2, "rule 2", RuleKind.VULNERABILITY),
+                RuleFactory.createRule(1, "rule 1", RuleKind.CODE_SMELL, BALLERINAX_ORG, "exampleModule"),
+                RuleFactory.createRule(3, "rule 3", RuleKind.BUG, BALLERINA_ORG, "exampleModule"),
+                RuleFactory.createRule(2, "rule 2", RuleKind.VULNERABILITY, BALLERINAX_ORG, "exampleModule"),
+                RuleFactory.createRule(1, "rule 1", RuleKind.CODE_SMELL, "wso2", "exampleModule"),
+                RuleFactory.createRule(3, "rule 3", RuleKind.BUG, "exampleOrg", "exampleModule"),
+                RuleFactory.createRule(2, "rule 2", RuleKind.VULNERABILITY, "exampleOrg", "exampleModule"),
+                RuleFactory.createRule(3, "rule 3", RuleKind.BUG, "wso2", "exampleModule"),
+                RuleFactory.createRule(2, "rule 2", RuleKind.VULNERABILITY, BALLERINA_ORG, "exampleModule"),
+                RuleFactory.createRule(1, "rule 1", RuleKind.CODE_SMELL)
+        ));
+        ScanUtils.sortRules(rules);
+        Assert.assertEquals(rules.get(0).id(), "ballerina:1");
+        Assert.assertEquals(rules.get(1).id(), "ballerina:2");
+        Assert.assertEquals(rules.get(2).id(), "ballerina:3");
+        Assert.assertEquals(rules.get(3).id(), "ballerina/exampleModule:1");
+        Assert.assertEquals(rules.get(4).id(), "ballerina/exampleModule:2");
+        Assert.assertEquals(rules.get(5).id(), "ballerina/exampleModule:3");
+        Assert.assertEquals(rules.get(6).id(), "ballerinax/exampleModule:1");
+        Assert.assertEquals(rules.get(7).id(), "ballerinax/exampleModule:2");
+        Assert.assertEquals(rules.get(8).id(), "ballerinax/exampleModule:3");
+        Assert.assertEquals(rules.get(9).id(), "wso2/exampleModule:1");
+        Assert.assertEquals(rules.get(10).id(), "wso2/exampleModule:2");
+        Assert.assertEquals(rules.get(11).id(), "wso2/exampleModule:3");
+        Assert.assertEquals(rules.get(12).id(), "exampleOrg/exampleModule:1");
+        Assert.assertEquals(rules.get(13).id(), "exampleOrg/exampleModule:2");
+        Assert.assertEquals(rules.get(14).id(), "exampleOrg/exampleModule:3");
+    }
```

Comment:
Shall we use a data-provider function to parse the arguments?

Check this out:

https://github.com/ballerina-platform/ballerina-lang/blob/master/tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/statements/SymbolsInCallStatementsTest.java#L60-L86